test/e2e/reconciler/namespace/controller_test.go

```diff
@@ -158,20 +166,51 @@ func TestNamespaceScheduler(t *testing.T) {
 					GenerateName: "e2e-nss-2-",
 				},
 				Spec: clusterv1alpha1.ClusterSpec{
-					KubeConfig: string(server2Kubeconfig),
+					KubeConfig:    string(server2Kubeconfig),
+					Unschedulable: true, // cordoned.
 				},
 			}, metav1.CreateOptions{})
 			require.NoError(t, err, "failed to create cluster2")
 
-			t.Log("Delete the old cluster and expect the namespace to end up scheduled to the new cluster.")
+			t.Log("Delete the old cluster and expect the namespace to end up unschedulable (since the new cluster is cordoned).")
 
 			err = clusterClient.Delete(ctx, cluster1.Name, metav1.DeleteOptions{})
 			require.NoError(t, err, "failed to delete cluster1 %q", cluster1.Name)
 
+			err = expect(namespace, unschedulableMatcher())
+			require.NoError(t, err, "did not see namespace marked unschedulable")
+
+			t.Log("Uncordon the new cluster and expect the namespace to end up scheduled to it.")
+
+			schedulablePatchBytes := []byte(`[{"op":"replace","path":"/spec/unschedulable","value":false}]`) // uncordoned.
+			cluster2, err = clusterClient.Patch(ctx, cluster2.Name, types.JSONPatchType, schedulablePatchBytes, metav1.PatchOptions{})
+			require.NoError(t, err, "failed to uncordon cluster2")
```

Comment:
(No action required) Maybe wait for cluster2 to be marked ready?

Reply:
Yeah, I believe your commit had it wait to become Ready here, but I think that was actually unnecessary -- at least while it only considers the `Ready` condition, and not the value of `.spec.unschedulable`.

Setting `unschedulable=true` (or `evictAfter` in the past) doesn't remove the `Ready` condition (as of right now), though perhaps it should?

Reply:
Not necessary, just defensive. If the test doesn't check that a cluster is ready, there is a chance the agent responsible for setting the ready condition might not have done so and the scheduled check would time out rather than failing for a detectable reason.